src/build_queue.rs

```diff
@@ -291,9 +291,25 @@ impl BuildQueue {
     pub fn get_new_crates(&self, index: &Index) -> Result<usize> {
         let mut conn = self.db.get()?;
         let diff = index.diff()?;
-        let (mut changes, oid) = diff.peek_changes_ordered()?;
+
+        let Some(last_seen_reference) = self.last_seen_reference()? else {
+            // we should always have a last seen reference in the database, other than when
+            // initialising a new deployment (e.g. dev/test/staging), in those cases we don't want
+            // to rebuild the entire world, so we get crates-index-diff to fetch the current state
+            // then use the current head as the base and will only start building new crates from
+            // now on
+            let (_, oid) = diff.peek_changes_ordered()?;
```

Comment:
@Byron this is an edge-case, but it could be useful to have a method that does the fetching and working out the commit ids but doesn't actually compute the changes between them (basically all of `peek_changes_with_options` except the last line).

Reply:
If you are not interested in the actual changes, I think an immediate improvement to performance should be to use `peek_changes()` instead. That way, the same `oid` is produced in much less time - I'd be surprised if it takes longer than 2 seconds to compute which seems OK for an edge case.

Would that work for you?

Reply:
Still seems to take at least 10s even in release mode if there's no prior last-seen-reference (since it computes the entire changeset), but 🤷 this is such a rare operation and that is dwarfed by the time to clone the repo so it seems fine.